edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -228,23 +221,37 @@ impl ModuleRuntime for DockerModuleRuntime {
     type SystemInfoFuture = Box<dyn Future<Item = CoreSystemInfo, Error = Self::Error> + Send>;
     type RemoveAllFuture = Box<dyn Future<Item = (), Error = Self::Error> + Send>;
 
-    fn init(&self) -> Self::InitFuture {
+    fn init(&mut self, settings: Settings) -> Self::InitFuture {
         info!("Initializing module runtime...");
 
-        let created = self.network_id.clone().map_or_else(
-            || future::Either::B(future::ok(())),
-            |id| {
-                let filter = format!(r#"{{"name":{{"{}":true}}}}"#, id);
-                let client_copy = self.client.clone();
-                let fut = self
-                    .client
+        if let DockerModuleRuntime::Initialized(_) = *self {
+            return Box::new(Err(Error::from(ErrorKind::AlreadyInitialized)).into_future());
+        }
+
+        // Clippy incorrectly flags the use of `.map(..).unwrap_or_else(..)` code as being replaceable
+        // with `.ok().map_or_else`. This is incorrect because `.ok()` will result in the error being dropped.
+        // So we suppress this lint. There's an open issue for this on the Clippy repo:
+        //      https://github.com/rust-lang/rust-clippy/issues/3730
+        #[allow(clippy::result_map_unwrap_or_else)]
+        let created = init_client(settings.moby_runtime().uri())
+            .map(move |client| {
+                // update our state to reflect that we are initialized
+                *self = DockerModuleRuntime::Initialized(client.clone());
+
+                let network_id = settings.moby_runtime().network().to_string();
+                info!("Using runtime network id {}", network_id);
+
+                let filter = format!(r#"{{"name":{{"{}":true}}}}"#, network_id);
+                let client_copy = client.clone();
```

Comment:
Do you need this clone? `client` isn't used after line 247 so it should be able to be moved into the `and_then` closure.